hr_timesheet_sheet/models/hr_timesheet_sheet.py

```diff
@@ -481,21 +516,71 @@ def clean_timesheets(self, timesheets):
         return timesheets
 
     def delete_empty_lines(self, delete_empty_rows=False):
-        for name in self.line_ids.mapped('value_y'):
+        name_list = list(set(self.line_ids.mapped('value_y')))
+        for name in name_list:
             row = self.line_ids.filtered(lambda l: l.value_y == name)
             if row:
+                row_0 = fields.first(row)
                 ts_row = self.timesheet_ids.filtered(
-                    lambda x: x.project_id.id == row[0].project_id.id
-                    and x.task_id.id == row[0].task_id.id
+                    lambda x: x.project_id.id == row_0.project_id.id
+                    and x.task_id.id == row_0.task_id.id
                 )
-                if delete_empty_rows and self.add_line_project_id:
+                is_add_line = self.add_line_project_id == row_0.project_id \
+                    and self.add_line_task_id == row_0.task_id
+                if delete_empty_rows and is_add_line:
                     check = any([l.unit_amount for l in row])
                 else:
                     check = not all([l.unit_amount for l in row])
                 if check:
                     ts_row.filtered(
                         lambda t: t.name == empty_name and not t.unit_amount
                     ).unlink()
+                    self.with_context(in_write=True).write(
+                        {'timesheet_ids': [
+                            (6, 0, self.timesheet_ids.exists().ids)]})
+                    if not ts_row.exists() and delete_empty_rows:
+                        self.with_context(in_write=True).write(
+                            {'line_ids': [
+                                (6, 0, (self.line_ids.exists() - row).ids)]})
+
+    def update_timesheets_according_lines(self):
+        """This method updates timesheets according
+        the values of the timesheet lines."""
+        if not self.id or self.add_line_project_id or self.add_line_task_id:
+            return
+        for line in self.line_ids:
+            sheet = line.sheet_id
+            timesheets = sheet.timesheet_ids.filtered(
+                lambda t: t.date == line.date
+                and t.project_id.id == line.project_id.id
+                and t.task_id.id == line.task_id.id
+            )
+            new_ts = timesheets.filtered(lambda t: t.name == empty_name)
+            amount = sum(t.unit_amount for t in timesheets)
+            diff_amount = line.unit_amount - amount
+            if len(new_ts) > 1:
+                new_ts = new_ts.merge_timesheets()
+                sheet.with_context(in_write=True).write(
+                    {'timesheet_ids': [
+                        (6, 0, sheet.timesheet_ids.exists().ids)]})
+            if not diff_amount:
+                continue
+            if new_ts:
+                unit_amount = new_ts.unit_amount + diff_amount
+                if unit_amount:
```

Comment:
@mreficent here needed ` if unit_amount:`??

I think you can directly write `new_ts.write({'unit_amount': unit_amount or 0.0 })`

Reply:
yeah, but then you will have the `else` becoming `if not unit_amount:`, thus the check is done inevitably.